node/rpc/src/chain_api.rs

```diff
@@ -323,3 +323,31 @@ where
     let name: String = data.state_manager.chain_config().name.clone();
     Ok(name)
 }
+
+// This is basically a port of the go implementation
```

Comment:
Note that there is another go implementation (Venus). Feel free to add in a comment a link to the method/commit that has "inspired" you.